tuist/tuist
# Fix tuist cache warm --xcframeworks of SPM frameworks due to missing macCatalyst availability

Resolves https://github.com/tuist/tuist/issues/5230

### Short description 📝

`tuist cache warm --xcframeworks` can't build XCFramework if SPM package does not explicitly specify `macCatalyst` availability. For example, `FirebaseCrashlytics` can't be built by `tuist > 3.18`.

This PR fixes this.

`macCatalyst` should be always available, because platforms in SPM package description can't be used to restrict supported platforms, it only allows customizing the deployment target versions:

>Restricting supported platforms: Some packages are only meant to work on certain platforms. This proposal doesn't provide ability to restrict the list of supported platforms and only allows customizing the deployment target versions. Restricting supported platforms is orthogonal to customizing deployment target and will be explored separately.

https://github.com/apple/swift-evolution/blob/main/proposals/0236-package-manager-platform-deployment-settings.md#future-directions

### How to test the changes locally 🧐

1. Download test project: https://github.com/tuist/tuist/files/11737174/TuistTest.zip
2. Clean cache: `swift run tuist clean --path path/to/downloaded/project`
3. Fetch dependencies: `swift run tuist fetch --path path/to/downloaded/project` // code changes will apply on this step
3. Build xcframework: `swift run tuist cache warm --xcframeworks --path path/to/downloaded/project`

It should successfully build.

### Contributor checklist ✅


### Reviewer checklist ✅
